modules/plink2/extract/main.nf

```diff
@@ -0,0 +1,38 @@
+process PLINK2_EXTRACT {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::plink2=2.00a2.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/plink2:2.00a2.3--h712d239_1' :
+        'quay.io/biocontainers/plink2:2.00a2.3--h712d239_1' }"
+
+    input:
+    tuple val(meta), path(pgen), path(psam), path(pvar), path(variants)
+
+    output:
+    tuple val(meta), path("*.pgen"), emit: pgen
+    tuple val(meta), path("*.psam"), emit: psam
+    tuple val(meta), path("*.pvar"), emit: pvar
+    path "versions.yml"            , emit: versions
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if( "$pgen" == "${prefix}.pgen" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
+    """
+    plink2 \\
+        --pfile ${pgen.baseName} \\
+        $args \\
+        --threads $task.cpus \\
+        --extract $variants \\
+        --make-pgen \\
+        --out ${prefix}
```

Comment:
```suggestion
        --out ${prefix}
```

I'm not familiar with the tool, can downstream tools accept gzipped output from `plink2`? If not don't worry, if yes it woudl be great to have a gzip command after this one to compress all the output for publishing.

Reply:
`plink2` natively supports [zstandard compression](http://www.cog-genomics.org/plink/2.0/other#zstd) :thinking: